File: app/src/main/java/com/example/android/newsapp/refactor/repository/RepositoryProvider.java
package com.example.android.newsapp.refactor.repository;

import com.example.android.newsapp.refactor.assemble.RemoteModule;
import com.example.android.newsapp.refactor.repository.remote.NewsRemoteRepository;

public class RepositoryProvider {

    public static NewsGateway provideNewsRepository() {
        return new NewsRemoteRepository(RemoteModule.getService());
    }
}

File: app/src/main/kotlin/com/example/android/newsapp/repository/RepositoryProvider.kt
package com.example.android.newsapp.repository

import com.example.android.newsapp.assemble.RemoteModule
import com.example.android.newsapp.repository.remote.NewsRemoteRepository


object RepositoryProvider {

    fun provideNewsRepository(): NewsGateway {
        return NewsRemoteRepository(RemoteModule.assembleService())
    }
}
